src/naksu/boxversion/boxversion.go

```diff
@@ -0,0 +1,202 @@
+package boxversion
+
+import (
+  "os"
+  "fmt"
+  "time"
+  "errors"
+  "regexp"
+  "io/ioutil"
+  "path/filepath"
+
+  "naksu/constants"
+  "naksu/mebroutines"
+  "naksu/network"
+  "naksu/xlate"
+)
+
+// Cache for GetVagrantBoxAvailVersionDetails
+type lastBoxAvail struct {
+  boxString string
+  boxVersion string
+  boxTimestamp int64
+  updateStarted int64
+}
+
+// Global cache for GetVagrantBoxAvailVersionDetails()
+var vagrantBoxAvailVersionDetailsCache lastBoxAvail
+var vagrantBoxAvailVersionDetailsCacheTimeout int64 = 5*60	// in seconds (5 minutes)
+
+
+// GetVagrantFileVersion returns a human-readable localised version string
+// for a given Vagrantfile (with "" defaults to ~/ktp/Vagrantfile)
+func GetVagrantFileVersion (vagrantFilePath string) string {
+	if vagrantFilePath == "" {
+			vagrantFilePath = mebroutines.GetVagrantDirectory() + string(os.PathSeparator) + "Vagrantfile"
```

Comment:
I'd use `filepath.Join(...)` for this. https://golang.org/pkg/path/filepath/#Join

Reply:
:1st_place_medal: There are great number of similar expressions. I think it would be good idea to use `filepath.Join()` for all of them.